### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ryanhall07**

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -70,11 +70,11 @@ ext.dep = [
   "moshiKotlin": "com.squareup.moshi:moshi-kotlin:1.9.2",
   "mysql": "mysql:mysql-connector-java:8.0.16",
   "nettyBom": "io.netty:netty-bom:4.1.34.Final",
-  "okHttp": "com.squareup.okhttp3:okhttp:4.6.0",
-  "okHttpMockWebServer": "com.squareup.okhttp3:mockwebserver:4.6.0",
+  "okHttp": "com.squareup.okhttp3:okhttp:4.5.0",
```
[the remark this answers is not in the document]
Comment (ryanhall07) [reply dependencies.gradle:73]: upgrading our master deps to 4.6 

was added here https://github.com/cashapp/misk/pull/1502

File: dependencies.gradle [hunk 2]
```diff
@@ -85,14 +85,14 @@ ext.dep = [
...
+  "retrofit": "com.squareup.retrofit2:retrofit:2.8.1",
+  "retrofitMoshi": "com.squareup.retrofit2:converter-moshi:2.8.1",
+  "retrofitWire": "com.squareup.retrofit2:converter-wire:2.8.1",
   "servletApi": "javax.servlet:javax.servlet-api:3.1.0",
   "slf4jApi": "org.slf4j:slf4j-api:1.7.28",
   "tink": "com.google.crypto.tink:tink:1.2.0",
-  "tracingDatadog": "com.datadoghq:dd-trace-api:0.46.0",
-  "tracingJaeger": "io.jaegertracing:jaeger-core:1.1.0",
+  "tracingDatadog": "com.datadoghq:dd-trace-api:0.49.0",
+  "tracingJaeger": "com.uber.jaeger:jaeger-core:0.27.0",
```
[the remark this answers is not in the document]
Comment (ryanhall07) [reply dependencies.gradle:95]: upgrading our master deps to match

was added here https://github.com/cashapp/misk/pull/1395

File: dependencies.gradle [hunk 3]
```diff
@@ -55,7 +55,7 @@ ext.dep = [
   "kotlinGradlePlugin": "org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.71",
   "kotlinNoArgPlugin": "org.jetbrains.kotlin:kotlin-noarg:1.3.71",
   "kotlinReflection": "org.jetbrains.kotlin:kotlin-reflect:1.3.71",
-  "kotlinStdLib": "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.3.71",
+  "kotlinStdLib": "org.jetbrains.kotlin:kotlin-stdlib:1.3.71",
```
[the remark this answers is not in the document]
Comment (ryanhall07) [reply dependencies.gradle:58]: this needs to change to `kotlinStdLibJdk8`

[the remark this answers is not in the document]
Comment (swankjesse) [reply dependencies.gradle:58]: Yep

File: dependencies.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (swankjesse) [reply dependencies.gradle:73]: Yeah, going forwards should be safe.  4.7.2 !

File: dependencies.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (swankjesse) [reply dependencies.gradle:95]: We should delete Jaeger!

**Comment by swankjesse** [comment]
Thanks!

File: dependencies.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (ryanhall07) [reply dependencies.gradle:95]: yes we should. i'll leave that for another time

File: dependencies.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryanhall07) [reply dependencies.gradle:73]: k. but i'll do it the right way of upgrading polyrepo first :)

File: dependencies.gradle [hunk 4]
```diff
@@ -4,46 +4,46 @@ ext.dep = [
...
   "curatorFramework": "org.apache.curator:curator-framework:4.0.1",
   "datasourceProxy": "net.ttddyy:datasource-proxy:1.4.9",
   "docker": "com.github.docker-java:docker-java:3.1.5",
-  "gcpCloudStorage": "com.google.cloud:google-cloud-storage:1.82.0",
-  "gcpDatastore": "com.google.cloud:google-cloud-datastore:1.82.0",
+  "gcpCloudStorage": "com.google.cloud:google-cloud-storage:1.108.1",
+  "gcpDatastore": "com.google.cloud:google-cloud-datastore:1.108.1",
   "gcpKms": "com.google.apis:google-api-services-cloudkms:v1-rev20190626-1.30.1",
-  "gcpLogback": "com.google.cloud:google-cloud-logging-logback:0.116.0-alpha",
+  "gcpLogback": "com.google.cloud:google-cloud-logging-logback:0.39.0-alpha",
```
[the remark this answers is not in the document]
Comment (ryanhall07) [reply dependencies.gradle]: upgrading our master deps to match

**Pull request merged**